docs/how-tos/use-multi-format-reader.md

```diff
@@ -0,0 +1,396 @@
+```python
+def get_attr(self, key: str, path: str) -> Any:
+    """
+    Get the metadata that was stored in the main file.
+    """
+    if self.hdf5_data is None:
+        return None
+    
+    return self.hdf5_data.get(path)
+```
+Here, we are making use of the `path`, which is the part of the config value that comes after the `@attrs:` prefix. For example, for the config value `"@attrs:metadata/instrument/version"`, the extracted path is `metadata/instrument/version`, which is also one of the keys of the `self.hdf5_data` dictionary.
+
+For the ELN data, we must implement the `get_eln_data` function that gets called from the `eln_callback` when using the `@eln` prefix:
+```python
+def get_eln_data(self, key: str, path: str) -> Any:
+    """Returns data from the given eln path."""
+    if self.eln_data is None:
+        return None
+    
+    return self.eln_data.get(key)
+```
+Here, we are making use of the fact that we have used `CONVERT_DICT` in the `parse_yml` function above. Thus, the keys of the `self.eln_data` dictionary are exactly the same as those in the config file. We can therefore just get this data using the `key` coming from the config file.
+
+Finally, we also need to adddress the `@data` prefix, which gets used in the `data_callback` to populate the NXdata group in the template. Note that here we use the same `@data` prefix to fill the `x_values` as well as the `data` (from `y_values`) fields. We achieve this by using the path that follows `@data:` in the config file:
+```python
+def get_data(self, key: str, path: str) -> Any:
+    """Returns measurement data from the given hdf5 path."""
+    if path.endswith(("x_values", "y_values")):
+        return self.hdf5_data.get(f"data/{path}")
+    else:
+        logger.warning(f"No axis name corresponding to the path {path}.")
+```
+
+## Bringing it all together
+Et voilà! That's all we need to read in our data and populate the `NXsimple` template. Our final reader looks like this:
+
+```python
+import logging
+from typing import Dict, Any
+import h5py
+
+from pynxtools.dataconverter.readers.multi.reader import MultiFormatReader
+from pynxtools.dataconverter.readers.utils import parse_yml
+
+logger = logging.getLogger("pynxtools")
+
+CONVERT_DICT = {
+    "unit": "@units",
+    "version": "@version",
+    "user": "USER[user]",
+    "instrument": "INSTRUMENT[instrument]",
+    "detector": "DETECTOR[detector]",
+    "sample": "SAMPLE[sample]",
+}
+
+
+class MyDataReader(MultiFormatReader):
+    """MyDataReader implementation for the DataConverter to convert mydata to NeXus."""
+    
+    supported_nxdls = [
+        "NXsimple"
+    ]
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.extensions = {
+            ".yml": self.handle_eln_file,
+            ".yaml": self.handle_eln_file,
+           ".json": self.set_config_file,
+            ".hdf5": self.handle_hdf5_file,

```

Comment:
Maybe a note of caution should be added:
along these lines:
"Case selection logics that rely on the filename suffix ! (not even the binary signature, mimetype) can be insufficient when multiple versions of hdf5 files exists, typical example H5OINA is HDF5 but there is v1, v2, v3 - v6... many concepts intersect but many are different

Reply:
Yeah, the idea is that the `self.handle_hdf5_file` method would implement such case selection logic. I will add a disclaimer in the `learn` docs.